core/src/main/java/kafka/server/ReplicaFetcherTierStateMachine.java

```diff
@@ -229,6 +230,9 @@ private Long buildRemoteLogAuxState(TopicPartition topicPartition,
                 Partition partition = replicaMgr.getPartitionOrException(topicPartition);
                 partition.truncateFullyAndStartAt(nextOffset, false, Option.apply(leaderLogStartOffset));
 
+                brokerTopicStats.topicStats(topicPartition.topic()).buildRemoteLogAuxStateRequestRate().mark();
+                brokerTopicStats.allTopicsStats().buildRemoteLogAuxStateRequestRate().mark();
```

Comment:
Instead, we could call `replicaMgr.brokerTopicStats().topicStats(...)` here directly to avoid a null pointer exception in the constructor.